File: src/main/java/VetCare/Back/model/entities/Agendamento.java
package VetCare.Back.model.entities;

import VetCare.Back.model.enuns.EnumStatusAgendamento;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "agendamentos")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Agendamento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pet_id", nullable = false)
    private Pet pet;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "veterinario_id", nullable = false)
    private Veterinario veterinario;

    private LocalDateTime dataHora;
    private String observacoes;

    @Enumerated(EnumType.STRING)
    private EnumStatusAgendamento status = EnumStatusAgendamento.AGENDADO;
}

File: front/app/types/agendamento.ts
export type EnumStatusAgendamento = 'AGENDADO' | 'CONFIRMADO' | 'CANCELADO' | 'REALIZADO';

export interface AgendamentoRequest {
  dataHora: string;
  observacoes?: string;
  status?: EnumStatusAgendamento;
  pet: { id: number };
}

export interface AgendamentoResponse {
  id: number;
  petId: number;
  veterinarioId: number;
  dataHora: string;
  observacoes: string;
  status: EnumStatusAgendamento;
}
